inst/fill/fill.css

```diff
@@ -1,19 +1,21 @@
-.html-fill-container {
-  display: flex;
-  flex-direction: column;
-  /* Prevent the container from expanding vertically or horizontally beyond its
-     parent's constraints. */
-  min-height: 0;
-  min-width: 0;
-}
-.html-fill-container > .html-fill-item {
-  /* Fill items can grow and shrink freely within
-     available vertical space in fillable container */
-  flex: 1 1 auto;
-  min-height: 0;
-  min-width: 0;
-}
-.html-fill-container > :not(.html-fill-item) {
-  /* Prevent shrinking or growing of non-fill items */
-  flex: 0 0 auto;
+@layer htmltools {
```

Comment:
I like the naming/intention here. And since we want this layer to potentially include other things beyond fill, it would be nice to have that more obvious from the code. In other words, have a file like `inst/htmltools.css` that contains:

```css
@layer htmltools {
    @import url("fill/fill.css")
}
```

(Although, to help make the result a standalone file, it'd probably make more sense for it to be a Sass file?)

And, of course, if we do this, we'd also want to update the corresponding `HTMLDependency()`

Reply:
Also, since this is such a far reaching feature, and `@layer` is still only ~94%, it'd be great to [provide a fallback](https://css-tricks.com/css-cascade-layers/#aa-query-feature-support-using-supports). I think it'd be something like?

```css
@supports at-rule(@layer) {
  @layer htmltools {
     @import url("fill/fill.css")
  }
}

@supports not at-rule(@layer) {
  @import url("fill/fill.css")
}
```

Reply:
I don't think we need to define the `htmltools` layer all at once. If we have other files that want to add rules to the layer, they can just wrap their rules in `@layer htmltools`.

Here's a small example: https://stackblitz.com/edit/vitejs-vite-qphmes?file=index.html

Reply:
Neat, good to know. I still do think we should do `@supports` which means we'd want to `@import`? There'd also be added benefit of no extra indentation and extraneous git history. 

I'd be happy to pick this up if you're trying to focus on PyShiny theming

Reply:
Interestingly, cascade layers has [94.18% support](https://caniuse.com/css-cascade-layers) with across-the-board support since around April 2022.

`@supports at-rule(@layer)` on the other hand has [93.37% support](https://caniuse.com/mdn-css_at-rules_layer) with basically the same date-range profile.

So I'm not sure it's worth the effort to use `@supports` in this case. If do decide to use `@supports`, I would also strongly urge us to just use copy-paste in this case. `fill.css` is only three rules and a handful of properties long.

Reply:
Huh, are you sure that https://caniuse.com/mdn-css_at-rules_layer is referring to `@supports at-rule(@layer)`? The only link on that page links to the doc reference for `@layer`

I was assuming browser support for `@supports at-rule(@layer)` was at 97.48% via https://caniuse.com/css-supports-api

Reply:
Yeah, I'm not absolutely confident but I'm pretty sure I'm interpreting caniuse correctly. I was reading [this blog post](https://www.matuzo.at/blog/2022/100daysof-day68/) which was written back when cascade layers first had full browser support. It mentions that `@supports at-rule(@layer)` had barely any browser support at that time. If you search caniuse for `at-rule`, there is data for other `at-rule(@____)` declarations.

Reply:
Ah, ok, I'm good with this then.